src/analytics/ValoraAnalytics.ts

```diff
@@ -86,33 +87,32 @@ async function getDeviceInfo(): Promise<DeviceInfoType> {
   }
 }
 
-const SEGMENT_OPTIONS: analytics.Configuration = {
-  using: [FIREBASE_ENABLED ? Firebase : undefined, Adjust, CleverTapSegment].filter(isPresent),
-  flushAt: 20,
-  debug: __DEV__,
-  trackAppLifecycleEvents: true,
-  recordScreenViews: true,
-  trackAttributionData: true,
-  ios: {
-    trackAdvertising: false,
-    trackDeepLinks: true,
-  },
-}
-
 class ValoraAnalytics {
   sessionId: string = ''
   deviceInfo: DeviceInfoType | undefined
 
   private currentScreenId: string | undefined
   private prevScreenId: string | undefined
+  private segmentClient: SegmentClient | undefined
 
   async init() {
     let uniqueID
     try {
       if (!SEGMENT_API_KEY) {
         throw Error('API Key not present, likely due to environment. Skipping enabling')
       }
-      await Analytics.setup(SEGMENT_API_KEY, SEGMENT_OPTIONS)
+      this.segmentClient = createClient({
```

Comment:
@kathaypacific fyi this upgrade may impact your work to initialize Segment earlier..